Add viewing/editing/filtering of item reporting category

coalest opened the pull request:
Resolves #5011

### Description
Added viewing, editing, and filtering of item reporting_category

### Type of change
* Breaking change (fix or feature that would cause existing functionality to not work as expected)

Breaking change as you can no longer filter items by base item, only by reporting category. 

### How Has This Been Tested?
Added tests and manually tested.

### Screenshots

Item index:
![Screenshot from 2025-05-28 17-46-30](https://github.com/user-attachments/assets/e6512fd3-3bed-448f-8793-d3ab91dfeee8)

Item show:
![Screenshot from 2025-05-28 17-46-37](https://github.com/user-attachments/assets/d7830716-2d29-45b5-8e88-71ce16d226c3)

Item edit:
![Screenshot from 2025-05-28 17-46-52](https://github.com/user-attachments/assets/bdfa946c-a13b-45c0-bd9e-d0b1d5e1a1d4)

cielf commented:
@coalest Please change the label "Reporting Category" to "NDBN Reporting Category" throughout. It's a bit redundant, but I think we should make it explicit.    Once you've done that, please also update the user guide. 

Other than that, looks great functionally!

coalest commented:
@cielf Made the requested changes. Can you take a look at the documentation I wrote, to make sure it is accurate and makes sense?

cielf commented:
@coalest -- I tried this with the production data,  and I experienced an issue...    
Slacking you with the screenshots (because I don't want to take the time to screen them for anything sensitive)

cielf commented:
Further investigation reveals a data problem on prod -- some case that wasn't caught with the initial reporting category creation.   Will investigate further.   This will put a hold on merging this PR, but I don't see why we can't have @dorner do the technical review of this in the meantime.

coalest commented:
@cielf Made the requested updates:

- Updated the screenshots in this issue after the text update.
- Fixed the error for items without a reporting category.
- Disabled the reporting category field when editing items housing a kit.

One question regarding UX. It seems like it would be a nice feature, such that when a user is creating a new item, when they select a base item, the reporting_category is automatically updated according to NDBN rules. It can still be changed, but at least it would save the user some time and reduce mistakes. What do you think? Should I add that?

cielf commented:
Hrm.    I'll need to double check / ask @dorner   but I think this is the point where we can finally sever the connection to the base item.       Which would solve the issue above, yes?

Not working on HE today,  but should be able to look into it tomorrow.

cielf commented:
(Nods)  The intent is that we will still have the items set up from the suite of base items when there is a new organization,   but if they are entering a new item,  they will have to indicate the NDBN reporting category themselves.

cielf commented:
Hey @coalest -- Just checking in on this one!

coalest commented:
Sorry for the late reply. Thanks for the clarification! I will work on removing the reliance on base items

coalest commented:
Also when you click "Export items" to get a csv on the items index page. That currently returns the name of the base item in the csv. Do we want to change that to reporting category? Or leave it as is?

cielf commented:
Yup -- what we mean is that Item will no longer have a BaseItem.   

The base items will still exist, but they are *only* going to be used to populate a new organization's items.

So...   
Yes, it does mean that we can remove the base item from items in the UI.  

I had forgotten about the base item in the export -- we will need to communicate to the users about that so they can adjust appropriately,  but yep, remove it.

coalest commented:
Ok just removed base items column from the items csv export. This should be ready for review again.

cielf commented:
(nods)  For clarity -- when I said we can sever the link between item and BaseItem,  I think we can actually remove the "belongs_to"  base_item on Item.   

It is, of course, possible that this will break something!     

I'll go ahead and do the functional testing on what we've got here, though.

cielf commented:
This is going to take me a couple of sessions to get through the testing of, I'm afraid
I ran into something when I was testing the new bank creation that *looks* unrelated, but I'm going to have to recreate with the current production code, etc., etc.

cielf reviewed (dismissed):
Hey @coalest  -- It's looking pretty good,  but I found something that slipped through the reviews of the earlier parts.

"Menstrual"  shouldn't be a selectable NDBN reporting category -- it's conceptually a super category consisting of pads, period liners, period underwear, tampons, and period other.        

Also -- For the kits case,   could we add a note on the NDBN reporting category, when it's a kit,  to indicate that "Kits are reported based on their contents".   Just to mitigate any confusion.

cielf commented:
@dorner -- sorry, I neglected to indicate that  I think this is ready for the first pass of a technical review -- the two things I've noted are quite minor changes.

dorner reviewed (approved):
All looks good on my end!

cielf commented:
Ok @coalest  -- We still need those two minor fixes, but once those are in we'll be good to go.

cielf commented:
Hey @coalest -- are you able to finish this off (or should we get someone else to handle those last tweaks)?   Thanks!

dorner reviewed (approved):
Looks good!
